### DISCUSSION TITLE
Interception request to obtain JS upload binary file function

### DISCUSSION THREAD

**Discussion by wenbei421** [cefsharp/CefSharp#4134] [General]
I'm using cefsharp to host a web page that uses js to request a custom protocol Url and request interception in C#, but I can't get the binary content of the file。
js code:
```js
  function upload() {
            // Form Data
            var postData = new FormData();

            var fileSelect = document.getElementById("fileSelect");
            if (fileSelect.files && fileSelect.files.length == 1) {
                var file = fileSelect.files[0]
                postData.set("file", file, file.name);
            }

            axios.post('http://icreate.com/admincontroller/upload', postData)
                .then(response => {
                    console.log(response.data)
                })
                .catch(error => {
                    console.log(error)
                });
        }
```
C# code
```c#
public class DefaultRequestSchemeHandler : ResourceHandler
    {
      ...
     public override CefReturnValue ProcessRequestAsync(IRequest request, ICallback callback)
        {
             ...
             var postDataElements = request?.PostData?.Elements;
             ...
        }
}
```
post data：
![image](https://user-images.githubusercontent.com/6428597/172900924-3442825a-f03e-4dd4-a0f3-0b06f6cf16da.png)

**Comment by amaitland** [comment]
In future please use discussions to ask questions rather than the feature request. https://github.com/cefsharp/CefSharp/discussions

> CefSharp is a wrapper around the [Chromium Embedded Framework](https://bitbucket.org/chromiumembedded/cef/overview) in a lot of cases a feature must be added to CEF first before it can be used in CefSharp. CEF has its own Feature Request Forum at https://magpcss.org/ceforum/viewforum.php?f=7

As per https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/feature_request.md

Enhancements to the request interception API need to be made upstream first. 

> but I can't get the binary content of the file

You have the path to the file, what's stopping you from reading it?

**Reply by wenbei421** [reply]
The case may not be appropriate, but in a real scenario it might be that js creates a blob of data that needs to be sent to C# for processing, such as a PDF print that goes directly to the printer, where the PDF is generated by js

**Reply by amaitland** [reply]
You logic would differ based on the Type property, for type File you can read the file from disk. For Type Bytes you can read from the Bytes property.

**Reply by wenbei421** [reply]
The Bytes attribute was not the content of the file at the time, I sent tens of megabytes of files, this is only a few hundred Bytes

**Reply by amaitland** [reply]
As I said you'll need to read the file from disk yourself. You have access to the raw post data straight from chromium. At this point in time chromium has done very little processing. 

 https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.2

**Comment by wenbei421** [comment]
Can you provide some examples for me?
